src/main/java/com/ericsson/gerrit/plugins/eiffel/messaging/EiffelEventSender.java

```diff
@@ -1,25 +1,140 @@
+import java.net.URISyntaxException;
+
+import org.apache.commons.codec.binary.Base64;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.http.HttpHeaders;
+import org.apache.http.HttpStatus;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.ericsson.eiffelcommons.utils.HttpRequest;
+import com.ericsson.eiffelcommons.utils.ResponseEntity;
 import com.ericsson.gerrit.plugins.eiffel.configuration.EiffelPluginConfiguration;
 import com.ericsson.gerrit.plugins.eiffel.events.EiffelEvent;
+import com.ericsson.gerrit.plugins.eiffel.exceptions.HttpRequestFailedException;
+import com.ericsson.gerrit.plugins.eiffel.exceptions.MissingConfigurationException;
+import com.ericsson.eiffelcommons.utils.HttpRequest.HttpMethod;
 import com.google.gson.Gson;
-import com.google.gson.JsonObject;
 
 public class EiffelEventSender {
+    private static final String GENERATE_PUBLISH_ENDPOINT = "/generateAndPublish/";
+    private static final String MESSAGE_PROTOCOL = "mp";
+    private static final String MESSAGE_TYPE = "msgType";
+    private static final String EIFFEL_PROTOCOL = "eiffelsemantics";
+    private static final String CONTENT_TYPE_HEADER = "Content-Type";
+    private static final String APPLICATION_JSON = "application/json";
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(EiffelEventSender.class);
 
     private final Gson gson = new Gson();
-    private final EiffelPluginConfiguration pluginConfig;
+    private String eiffelMessage;
+    private String eiffelType;
+    private EiffelPluginConfiguration pluginConfig;
+    private HttpRequest httpRequest;
 
     public EiffelEventSender(EiffelPluginConfiguration pluginConfig) {
+        this(pluginConfig, new HttpRequest(HttpMethod.POST));
+    }
+
+    public EiffelEventSender(EiffelPluginConfiguration pluginConfig, HttpRequest httpRequest) {
+        this.httpRequest = httpRequest;
         this.pluginConfig = pluginConfig;
     }
 
-    public void send(EiffelEvent eiffelEvent) {
-        JsonObject eiffelEventJson = gson.toJsonTree(eiffelEvent).getAsJsonObject();
-        System.out.println("Event to send::: " + eiffelEventJson.toString());
-        System.out.println("REMReM URL ::: " + pluginConfig.getRemremPublishURL());
-        // TODO Implement event sending using REMReM details from pluginconfig.
+    /**
+     * Sends a REMReM eiffel message using
+     *
+     */
+    public void send() {
+        try {
+            verifyConfiguration();
+            generateAndPublish();
+        } catch (URISyntaxException | IOException | MissingConfigurationException
+                | HttpRequestFailedException e) {
+            LOGGER.error("Failed to send eiffel message.", e);
+        }
+    }
+
+    public EiffelEventSender setMessage(final EiffelEvent eiffelMessage) {
+        this.eiffelMessage = gson.toJsonTree(eiffelMessage).toString();
+        return this;
+    }
+
+    public EiffelEventSender setType(final String eiffelType) {
```

Comment:
Some Javadoc here.

Reply:
It is not clear to me what Type is.

Reply:
So "Set the eiffel event type", that is the sort of sentence that I think doesn't add anything useful. Do you have anything in mind for the javadoc? Perhaps the method could be called something different if you feel setType is not enough?

Reply:
That should have added a lot, but then i think the name is not good enough. Maybe the name should be, setEiffelEventType then?

Reply:
I named them more specifically now to avoid using javadoc.